packages/arb-token-bridge-ui/src/components/TransferPanel/TokenSearchUtils.ts

```diff
@@ -67,86 +67,88 @@ function tokenListsToSearchableTokenStorage(
   l1ChainId: string,
   l2ChainId: string
 ): SearchableTokenStorage {
-  return (
-    tokenLists
-      //
-      .reduce((acc: SearchableTokenStorage, tokenList: TokenListWithId) => {
-        tokenList.tokens.forEach(token => {
-          const address = token.address.toLowerCase()
-          const stringifiedChainId = String(token.chainId)
-
-          if (stringifiedChainId === l1ChainId) {
-            // The address is from an L1 token
-
-            if (typeof acc[address] === 'undefined') {
-              // First time encountering the token through its L1 address
-              acc[address] = {
-                ...token,
-                type: TokenType.ERC20,
-                l2Address: undefined,
-                tokenLists: []
-              }
-            } else {
-              // Token was already added to the map through its L2 token
-              acc[address] = { ...acc[address], address }
+  return tokenLists.reduce(
+    (acc: SearchableTokenStorage, tokenList: TokenListWithId) => {
+      tokenList.tokens.forEach(token => {
+        const address = token.address.toLowerCase()
+        const stringifiedChainId = String(token.chainId)
+        const accAddress = acc[address]
+
+        if (stringifiedChainId === l1ChainId) {
+          // The address is from an L1 token
+          if (typeof accAddress === 'undefined') {
+            // First time encountering the token through its L1 address
+            acc[address] = {
+              ...token,
+              type: TokenType.ERC20,
+              l2Address: undefined,
+              tokenLists: []
+            }
+          } else {
+            // Token was already added to the map through its L2 token
+            acc[address] = {
+              ...accAddress,
+              address
             }
+          }
 
-            const tokenLists = acc[address].tokenLists
+          // acc[address] was defined in the if/else above
+          const tokenLists = accAddress!.tokenLists
 
-            if (!tokenLists.includes(tokenList.bridgeTokenListId)) {
-              acc[address].tokenLists.push(tokenList.bridgeTokenListId)
-            }
-          } else if (stringifiedChainId === l2ChainId) {
-            // The token is an L2 token
+          if (!tokenLists.includes(tokenList.bridgeTokenListId)) {
+            accAddress!.tokenLists.push(tokenList.bridgeTokenListId)
```

Comment:
although objects are passed by reference, i'd still prefer using `acc[address]` here as we're altering a property in `acc[address]`

so i think we should stick to the original version

Reply:
Agree, I still need to add `!` as TS doesn't pick that `acc[address]` is always defined, because of the if/else above